### ISSUE TITLE
[Bug]: Very Slow Script Executing

### ISSUE THREAD

**Issue by Kyoukazaki** [GameServerManagers/LinuxGSM#4462]
### User story

Very slow script execution

### Game

palworld/minecraft

### Linux distro

Ubuntu 22.04

### Command

command: start, command: stop, command: restart, command: monitor, command: details

### Further information

Freshly installed ubuntu server without anything extra running
I only tried it with palworld and minecraft
but ANY command i execute is lagging/hanging/processes slow
./pwserver start/stop/restart takes about 6-10 minutes to execute
i was checking resources if the script actually doing anything but no, CPU was idle, drive was idle..
i dont know how to check what is wrong with the script

Before anyone asks
Yes i am a linux Novice

### Relevant log output

```shell
Jan 24 15:28:33.758 pwserver: START: PASS: Started LinuxGSM
Jan 24 15:28:33.861 pwserver: START: PASS: Started LinuxGSM
Jan 24 15:28:33.868 pwserver: START: INFO: LinuxGSM version: v24.1.1
Jan 24 15:28:33.869 pwserver: START: PASS: core_exit.sh exiting with code: 0
Jan 24 15:42:01.048 pwserver: STOP: WARN: Unable to get public IP address
Jan 24 15:44:12.124 pwserver: STOP: WARN: Unable to get public IP address
Jan 24 15:44:12.631 pwserver: STOP: INFO: Graceful: CTRL+c
Jan 24 15:44:15.870 pwserver: STOP: PASS: Graceful: CTRL+c: OK: 3 seconds
Jan 24 15:44:15.896 pwserver: STOP: INFO: LinuxGSM version: v24.1.1
Jan 24 15:44:15.898 pwserver: STOP: PASS: core_exit.sh exiting with code: 0
```

**Comment by MicLieg** [comment]
Apparently more users are experiencing this bug: https://discord.com/channels/127498813903601664/219535041468956673/1204753801899413524

**Comment by rxmario** [comment]
I'm experiencing issues with LinuxGSM script executing very slowly. Here's a summary of the troubleshooting steps @MicLieg and I taken:

### Connection Tests: 
The server is a dedicated machine rented from Hetzner. It was initially working fine until it started experiencing issues after a few restarts and backups.

**Ping Test**: A ping test to ip-api.com was successful, indicating that there isn't a DNS issue blocking the connection.
`PING ip-api.com (208.95.112.1) 56(84) bytes of data.
64 bytes from ip-api.com (208.95.112.1): icmp_seq=1 ttl=251 time=12.7 ms
...
14 packets transmitted, 14 received, 0% packet loss, time 13024ms`

**Curl Command Test**: A test using curl -s http://ip-api.com/json got stuck, suggesting the request to ip-api.com was not completing as expected. A verbose curl (curl -v http://ip-api.com/json) showed that the connection to ip-api.com timed out, indicating a potential network issue or block.
`* Trying 208.95.112.1:80...
...
connect to 208.95.112.1 port 80 failed: Connection timed out`

**Tracepath Test**: Running tracepath ip-api.com showed that the request left the Hetzner network but received no replies afterward, suggesting that ip-api.com might be blocking my server's IP address.
`1?: [LOCALHOST] pmtu 1500
...
7: ae-3.a02.londen14.uk.bb.gin.ntt.net 16.335ms
8: no reply
...
23: no reply`

**Conclusion**: The troubleshooting indicates that the problem likely lies with ip-api.com blocking my server's IP address, not with DNS or local network issues. It was suggested that I might resolve the issue by requesting a new IP address from Hetzner.

**Comment by irobot73** [comment]
> I'm experiencing issues with LinuxGSM script executing very slowly. Here's a summary of the troubleshooting steps @MicLieg and I taken:
> 
> ### Connection Tests:
> The server is a dedicated machine rented from Hetzner. It was initially working fine until it started experiencing issues after a few restarts and backups.
> 
> **Ping Test**: A ping test to ip-api.com was successful, indicating that there isn't a DNS issue blocking the connection. `PING ip-api.com (208.95.112.1) 56(84) bytes of data. 64 bytes from ip-api.com (208.95.112.1): icmp_seq=1 ttl=251 time=12.7 ms ... 14 packets transmitted, 14 received, 0% packet loss, time 13024ms`
> 
> **Curl Command Test**: A test using curl -s http://ip-api.com/json got stuck, suggesting the request to ip-api.com was not completing as expected. A verbose curl (curl -v http://ip-api.com/json) showed that the connection to ip-api.com timed out, indicating a potential network issue or block. `* Trying 208.95.112.1:80... ... connect to 208.95.112.1 port 80 failed: Connection timed out`
> 
> **Tracepath Test**: Running tracepath ip-api.com showed that the request left the Hetzner network but received no replies afterward, suggesting that ip-api.com might be blocking my server's IP address. `1?: [LOCALHOST] pmtu 1500 ... 7: ae-3.a02.londen14.uk.bb.gin.ntt.net 16.335ms 8: no reply ... 23: no reply`
> 
> **Conclusion**: The troubleshooting indicates that the problem likely lies with ip-api.com blocking my server's IP address, not with DNS or local network issues. It was suggested that I might resolve the issue by requesting a new IP address from Hetzner.

Can confirm the same:
1) Ping == OK
2) Verbose CURL == **FAILURE** [_curl -v http://ip-api.com/json_]

```
*   Trying 0.0.0.0:80...
* connect to 0.0.0.0 port 80 failed: Connection refused
*   Trying :::80...
* connect to :: port 80 failed: Connection refused
* Failed to connect to ip-api.com port 80 after 3 ms: Connection refused
* Closing connection 0
curl: (7) Failed to connect to ip-api.com port 80 after 3 ms: Connection refused
```

3) Verbose CURL (using VPN) == **OK** [_curl -v http://ip-api.com/json_]

```
*   Trying 208.95.112.1:80...
* Connected to ip-api.com (208.95.112.1) port 80 (#0)
> GET /json HTTP/1.1
> Host: ip-api.com
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Fri, 16 Feb 2024 20:13:00 GMT
< Content-Type: application/json; charset=utf-8
< Content-Length: 290
< Access-Control-Allow-Origin: *
< X-Ttl: 60
< X-Rl: 44
< 
* Connection #0 to host ip-api.com left intact
{"status":"success","country":"United States","countryCode":"US","region":"GA","regionName":"Georgia","city":"Atlanta","zip":"30303","lat":33.7557,"lon":-84.3884,"timezone":"America/New_York","isp":"M247 Europe","org":"M247 Ltd Atlanta",...
```

**Comment by MicLieg** [comment]
Can you verify If your Host hast some sorry of Firewall enabled? The Last Case was solved by disabling some Firewall Rules.

**Comment by irobot73** [comment]
> Can you verify If you or your host hast some sort of firewall enabled? The last case was solved by disabling some firewall rules.

Good call. Yes, I'm running PiHole (looks like one quick a few RULES links I use).  

It did/was catching the URL & blocking....now whitelisted.  Hope it's as easy a 'fix' for others

**Comment by kaueraal** [comment]
I have the same issue. They seem to block my residential IP block, I cannot even ping them. I have no DNS filter or similar and I verified that the correct ip is resolved via comparing it to several large public resolvers. From my VPS I can successfully use curl, which uses the same resolved IP address for the query.

I resolved the issue for me via overwriting the resolved ip in my dns server, setting it to 127.0.0.1 which refuses the connection. Other users with the same issue can edit their host file or similar.

Either way, the curl command should use a relatively short timeout.
